File: src/main/java/ticklock/controller/dto/PurchaseResponse.java
package ticklock.controller.dto;

public record PurchaseResponse(
        boolean success,
        String message,
        int remainingSeats
) {
    public static PurchaseResponse success(int remainingSeats) {
        return new PurchaseResponse(true, "구매 성공", remainingSeats);
    }

    public static PurchaseResponse failure(String message) {
        return new PurchaseResponse(false, message, -1);
    }
}

File: src/main/kotlin/ticklock/controller/dto/PurchaseResponse.kt
package ticklock.controller.dto

data class PurchaseResponse(
    val success: Boolean,
    val message: String,
    val remainingSeats: Int
) {
    companion object {
        @JvmStatic
        fun success(remainingSeats: Int): PurchaseResponse = PurchaseResponse(
            success = true,
            message = "구매 성공",
            remainingSeats = remainingSeats
        )

        @JvmStatic
        fun failure(message: String): PurchaseResponse = PurchaseResponse(
            success = false,
            message = message,
            remainingSeats = -1
        )
    }
}